tests/integration/apigateway/test_apigateway_api.py

```diff
@@ -69,6 +77,145 @@ def _factory(*args, **kwargs):
 
 
 class TestApiGatewayApi:
+    @pytest.mark.aws_validated
+    def test_invoke_test_method(self, create_rest_apigw, snapshot, aws_client):
+        snapshot.add_transformer(
+            KeyValueBasedTransformer(
+                lambda k, v: str(v) if k == "latency" else None, "latency", replace_reference=False
+            )
+        )
+        snapshot.add_transformer(
+            snapshot.transform.key_value("log", "log", reference_replacement=False)
+        )
+
+        api_id, _, root = create_rest_apigw(name="aws lambda api")
+
+        # Create the /pets resource
+        root_resource_id, _ = create_rest_resource(
+            aws_client.apigateway, restApiId=api_id, parentId=root, pathPart="pets"
+        )
+        # Create the /pets/{petId} resource
+        resource_id, _ = create_rest_resource(
+            aws_client.apigateway, restApiId=api_id, parentId=root_resource_id, pathPart="{petId}"
+        )
+        # Create the GET method for /pets/{petId}
+        create_rest_resource_method(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="GET",
+            authorizationType="NONE",
+            requestParameters={
+                "method.request.path.petId": True,
+            },
+        )
+        # Create the POST method for /pets/{petId}
+        create_rest_resource_method(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="POST",
+            authorizationType="NONE",
+            requestParameters={
+                "method.request.path.petId": True,
+            },
+        )
+        # Create the response for method GET /pets/{petId}
+        create_rest_api_method_response(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="GET",
+            statusCode="200",
+        )
+        # Create the response for method POST /pets/{petId}
+        create_rest_api_method_response(
+            aws_client.apigateway,
+            restApiId=api_id,
+            resourceId=resource_id,
+            httpMethod="POST",
+            statusCode="200",
+        )
+        # Create the integration to connect GET /pets/{petId} to a backend
+        integration_uri, _ = create_rest_api_integration(
```

Comment:
nit: do we need anything this method returns?

Reply:
nice catch, removed